src/argilla_sdk/records/_io/_datasets.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2024-present, Argilla, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict, List, TYPE_CHECKING, Union
+
+from datasets import Dataset
```

Comment:
This would fail if datasets package is not present
```suggestion
try:
    from datasets import Dataset
except ImportError:
    warnings.warn("Hugging Face datasets is not installed. Please install it using `pip install datasets`.")
    Dataset = None
```

we should deal with other places where we're using the datasets package. Maybe we should centralize the logic somewhere.

Reply:
I missed this sorry. 

> we should deal with other places where we're using the datasets package. Maybe we should centralize the logic somewhere.

We have two options really:
- Use a `from_datasets` method in `DatasetRecords` so that we can import then type there and not in the generic add/update methods.
- Add a bool function in the `_io._datasets` that types on a hd `Dataset` attribute or name.